common-utils/src/main/java/eu/arrowhead/common/intf/properties/validators/NotEmptySetValidator.java

```diff
@@ -0,0 +1,70 @@
+package eu.arrowhead.common.intf.properties.validators;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import eu.arrowhead.common.Utilities;
+import eu.arrowhead.common.exception.InvalidParameterException;
+import eu.arrowhead.common.intf.properties.IPropertyValidator;
+import eu.arrowhead.common.service.validation.name.NameNormalizer;
+import eu.arrowhead.common.service.validation.name.NameValidator;
+
+@Service
+public class NotEmptySetValidator implements IPropertyValidator {
+
+	//=================================================================================================
+	// members
+
+	@Autowired
+	private NameValidator nameValidator;
+
+	@Autowired
+	private NameNormalizer nameNormalizer;
+
+	private final String nameArg = "name";
+
+	private final Logger logger = LogManager.getLogger(getClass());
+
+	//=================================================================================================
+	// methods
+
+	//-------------------------------------------------------------------------------------------------
+	@Override
+	public Object validateAndNormalize(final Object propertyValue, final String... args) throws InvalidParameterException {
+		logger.debug("NotEmptySetValidator.validateAndNormalize started...");
+
+		if (propertyValue instanceof final Set<?> set) {
```

Comment:
you can put final there? I didn't know that. Please fix this in HttpOperationsValidator.